### PR TITLE
ORC-503. Add Maven Wrapper

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
### What changes were proposed in this pull request?

This PR aims to add `Maven Wrapper`.

### Why are the changes needed?

This will help ORC build and testing at the environment without Maven installation.

### How was this patch tested?

Manual.
```
cd java
./mvnw clean
```

This closes #454

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
cc @omalley , @wgtmac , @pgaref

File: java/mvnw [hunk 1]
```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env bash
```
Comment (dongjoon-hyun) [inline java/mvnw:1]: This file is a simplified version of Apache Spark mvn wrapper.
- https://github.com/apache/spark/blob/master/build/mvn

**Comment by pgaref** [comment]
Hey @dongjoon-hyun  thanks for looking at this!
Wrapper script LGTM --  however I believe we still have to replace the MVN build/test calls with MVNW right?
For instance:

- https://github.com/apache/orc/blob/master/.github/workflows/master.yml#L46
- https://github.com/apache/orc/blob/master/java/CMakeLists.txt#L49

And as we are only now going to depend on MVNW maybe even remove maven plugin installation from Dockerfiles completely?

- https://github.com/apache/orc/blob/master/docker/debian10/Dockerfile#L33

Another thing we have to check is if the  MAVEN_OPTS are actually propagated to the wrapper (which I believe they should):
https://github.com/apache/orc/blob/master/.github/workflows/master.yml#L22

**Comment by dongjoon-hyun** [comment]
Thank you so much for review, @pgaref . 

For the following, we don't need to use this in our CI or testing tool. This is for new environment.
> however I believe we still have to replace the MVN build/test calls with MVNW right?

For the following, that is not a scope of this PR. If some Docker images has a problem with old Maven versions, we had better migrate it step by step.
> And as we are only now going to depend on MVNW maybe even remove maven plugin installation from Dockerfiles completely?

The Apache Spark script already handles it like the following and uses it in Apache Spark GitHub Action job.
> Another thing we have to check is if the MAVEN_OPTS are actually propagated to the wrapper (which I believe they should): 
```bash
# Set any `mvn` options if not already present
export MAVEN_OPTS=${MAVEN_OPTS:-"$_COMPILE_JVM_OPTS"}
```

**Comment by dongjoon-hyun** [comment]
Merged to master!

**Pull request merged**